src/components/input/input.js

```diff
@@ -80,11 +80,20 @@ function mdInputContainerDirective($mdTheming, $parse) {
 
   var INPUT_TAGS = ['INPUT', 'TEXTAREA', 'SELECT', 'MD-SELECT'];
 
+  var HIDDEN_INPUT_SELECTORS = ':not([hidden=""])'
```

Comment:
I'm not sure I understand. We are checking for the siblings on lines 92 and 97. The problem is, those checks need to avoid certain attributes. Patching into querySelector is faster than returning all of them and checking the attributes for each one.

That said, the entire concept of `md-icon-left` and `md-icon-right` is only to reduce the CSS and has no value on the Javascript side. Even if we move it to CSS, it will still need this list of selectors. 😑